internal/metrics/recorder.go

```diff
@@ -83,7 +94,49 @@ func NewRecorder(register bool) *Recorder {
 		addonsCount:           addonsCount,
 		addonOperatorPaused:   addonOperatorPaused,
 		ocmAPIRequestDuration: ocmAPIReqDuration,
+		addonHealthInfo:       addonHealthInfo,
+	}
+}
+
+func (r *Recorder) RecordAddonHealthInfo(addon *addonsv1alpha1.Addon,
+	clusterID string) {
+
+	var (
+		// `healthStatus` defaults to unknown unless status conditions say otherwise
+		healthStatus = 2
+		healthReason = "Unknown"
+		healthCond   = meta.FindStatusCondition(addon.Status.Conditions,
+			addonsv1alpha1.Available)
+	)
+
+	if healthCond != nil {
+		healthReason = healthCond.Reason
+
+		switch healthCond.Status {
+		case metav1.ConditionFalse:
+			healthStatus = 0
+		case metav1.ConditionTrue:
+			healthStatus = 1
+		default:
+			healthStatus = 2
+		}
+
+	}
+
+	if clusterID == "" {
+		clusterID = "unknown"
```

Comment:
I would like to eliminate this case, because we really need this ID to make sense of these metrics :)
After checking with Amador, we should also report the "cluster external ID" and not the cluster ID we use to talk with OCM, which actually makes things a bit simpler :)

I have quickly moved the initial get of the cluster ID out of the AddonOperatorReconciler, so we can just inject that parameter at boot-up time:
https://github.com/openshift/addon-operator/pull/157

Reply:
LGTM-ed! Will rebase this PR once #157 is merged :)